### PR TITLE
[pdal] no absolute paths

### PR DESCRIPTION
**Pull request by autoantwort**
For #20469

### CODE REVIEW

**Comment by autoantwort** [comment]
Also depends on #20439

**Comment by dg0yt** [comment]
Don't modernize outdated pdal too much. There is #21275.

**Comment by autoantwort** [comment]
Ah sorry, I only changed vcpkg_configure_cmake to vcpkg_cmake_configure etc. and added some quotes. But I can revert this changes

File: ports/pdal/portfile.cmake [hunk 1]
```diff
@@ -47,6 +47,11 @@ vcpkg_install_cmake(ADD_BIN_TO_PATH)
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/pdal/cmake)
 vcpkg_copy_pdbs()
 
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/pkgconfig" "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig") # dependency gdal has no pkgconfig files
+
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/pdal/Dimension.hpp" "${SOURCE_PATH}" "")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/pdal/pdal_features.hpp" "#define PDAL_PLUGIN_INSTALL_PATH \"${CURRENT_PACKAGES_DIR}/bin\"" "")
```
Comment (dg0yt) [inline ports/pdal/portfile.cmake:53]: It is probaly safe to define PDAL_PLUGIN_INSTALL_PATH as `"."`.

File: ports/pdal/usage [hunk 2]
```diff
@@ -0,0 +1,14 @@
+The package pdal provides CMake targets:
+
+    find_package(PDAL CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE pdalcpp pdal_util pdal_arbiter pdal_kazhdan)
+
+Add the following cmake code to your CMakeLists.txt to create definitions for DCMTK
```
Comment (dg0yt) [inline ports/pdal/usage:6]: It is PDAL, not DCMTK.

File: ports/pdal/usage [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    find_package(PDAL CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE pdalcpp pdal_util pdal_arbiter pdal_kazhdan)
+
+Add the following cmake code to your CMakeLists.txt to create definitions for DCMTK
+    if ("${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}" STREQUAL "/")
+        message(WARNING "No vcpkg install dir found, PDAL definitions are not set")
+    else()
+        target_compile_definitions(main
+            PUBLIC
+                "PDAL_PLUGIN_INSTALL_PATH=${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/bin"
```
Comment (dg0yt) [inline ports/pdal/usage:12]: I doubt that `bin` is the right path for linux or osx. I would prefer to not add non-portable instructions.

File: ports/pdal/usage (hunk 2 above)
Comment (dg0yt) [reply ports/pdal/usage:6]: In addition, I think it would be better to handle PDAL_PLUGIN_INSTALL_PATH in the config files.

**Review by dg0yt — CHANGES_REQUESTED** [review]
How about putting this PR in draft state? I will take it as a check list for the PDAL update.

**Comment by autoantwort** [comment]
> It is probaly safe to define PDAL_PLUGIN_INSTALL_PATH as ".".

Ok good, then the `usage` file is not needed.

**Pull request closed without merging**